source/common/http/conn_manager_impl.cc

```diff
@@ -577,6 +580,14 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
 }
 
 ConnectionManagerImpl::ActiveStream::~ActiveStream() {
+  if (connection_manager_.codec_->protocol() < Protocol::Http2) {
+    // For HTTP/2 there are still some reset cases where details are not set.
+    // For HTTP/1 there shouldn't be any. Regression-proof this.
+    ASSERT(filter_manager_.streamInfo().responseCodeDetails().has_value());
```

Comment:
@snowp @alyssawilk I suspect there is some race here with onEvent(RemoteClose) and deferred delete. We may not be able to detect remote close if we log immediately at the end of the stream. Any thoughts?

Reply:
If we're going to move logs around, think I'd like this assert removed to before we log (we want to make sure details are set before stream info is shipped to access logs)

I don't think we should have to set DownstreamRemoteDisconnect in two places, and I'm not sure what race there would be.  If we get a normal end_stream set, the stream should finish inline and not be lingering around to pick up and I'd think any subsequent disconnects wouldn't be registered?

Reply:
I'm getting test failures if I don't update stream info in the destructor. Might be a test issue again. Is it possible to get onEvent(RemoteClose) once stream is scheduled to be defer-deleted? If not, then it is likely just another unrealistic test.

Reply:
yeah, I'd guess it's the tests being badly ordered but you should be able to check with gdb or envoy logs.

Reply:
What I mean with the race is the following sequence:
1. Socket is read ready, data read, calling onData() and and encodeHeaders() in HCM.
2. encodeHeaders() ends stream, triggers log().
3. Socket is remote-closed.

It seems that we need to process the file events before triggering log() but I'm not sure how to hold onData() for tht.

Reply:
ah, in that case in production I think the details would basically be set on encodeHeaders (like the router will when proxying from upstream, or the HCM will when sendLocalReply is called) so maybe switch to that?

Reply:
Thanks for explaining. That works indeed.